lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp

```diff
@@ -479,11 +480,13 @@ Value loadA(ConversionPatternRewriter &rewriter, Location loc, Value thread,
             for (int elemId = 0; elemId < elemsPerLoad; ++elemId) {
               Value elemVal =
                   extract_element(aElemTy, vectorValue, i32_val(elemId));
+              elemVal = bitcast(elemVal, resElemTy);
               valVec = insert_element(vecTy, valVec, elemVal,
                                       i32_val(loadId * elemsPerLoad + elemId));
             }
           } else {
             valVec = extract_element(aElemTy, vectorValue, i32_val(0));
+            valVec = bitcast(valVec, resElemTy);
```

Comment:
Note: this part is repeated 4 times: for operand A/B * for fast/normal path

This is what I want to refactor in mfma dot...